vnext/Microsoft.ReactNative.Managed/ReactTaskExtensions.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.ReactNative.Managed
+{
+  public static class ReactTaskExtensions
+  {
+    public static Task ContinueWith(this Task task, IJSValueWriter writer, MethodResultCallback resolve,
+      MethodResultCallback reject)
+    {
+      return task.ContinueWith(t =>
+      {
+        if (t.IsCompletedSuccessfully)
+        {
+          if (resolve != null)
+          {
+            writer.WriteArgs();
+            resolve(writer);
+          }
+        }
+        else
+        {
+          if (reject != null)
+          {
+            var exception = t.Exception.Flatten();
+            WriteReject(writer, exception);
+            reject(writer);
+          }
+        }
+      });
+    }
+
+    public static Task ContinueWith<T>(this Task<T> task, IJSValueWriter writer, MethodResultCallback resolve,
+      MethodResultCallback reject)
+    {
+      return task.ContinueWith(t =>
```

Comment:
>ContinueWith [](start = 18, length = 12)

Is it possible to specify a thread or queue where this code is executed?

Reply:
it is, the default will be to use the TaskScheduler.Current with [ContinuationOptions.None](http://index/?leftProject=mscorlib&leftSymbol=uoa2pyfub43w&file=system%5Cthreading%5CTasks%5CTask.cs&line=6919)
```cs
        /// <summary>
        /// Default = "Continue on any, no task options, run asynchronously"
        /// Specifies that the default behavior should be used.  Continuations, by default, will
        /// be scheduled when the antecedent task completes, regardless of the task's final <see
        /// cref="System.Threading.Tasks.TaskStatus">TaskStatus</see>.
        /// </summary>
        None = 0,
```

See: [Code](http://index/?leftProject=mscorlib&leftSymbol=dooyykhhqojp&file=system%5Cthreading%5CTasks%5CTask.cs&line=3752)

Reply:
Added a comment to this effect